pkg/yamlmeta/document_set.go

```diff
@@ -26,32 +26,51 @@ func NewDocumentSetFromBytes(data []byte, opts DocSetOpts) (*DocumentSet, error)
 	return docSet, nil
 }
 
-func (n *DocumentSet) Print(writer io.Writer) {
-	NewPrinter(writer).Print(n)
+func (ds *DocumentSet) Print(writer io.Writer) {
+	NewPrinter(writer).Print(ds)
 }
 
 // AsSourceBytes() returns bytes used to make original DocumentSet.
 // Any changes made to the DocumentSet are not reflected in any way
-func (n *DocumentSet) AsSourceBytes() ([]byte, bool) {
-	if n.originalBytes != nil {
-		return *n.originalBytes, true
+func (ds *DocumentSet) AsSourceBytes() ([]byte, bool) {
+	if ds.originalBytes != nil {
+		return *ds.originalBytes, true
 	}
 	return nil, false
 }
 
-func (n *DocumentSet) AsBytes() ([]byte, error) {
-	return n.AsBytesWithPrinter(nil)
+func (ds *DocumentSet) AsBytes() ([]byte, error) {
+	return ds.AsBytesWithPrinter(nil)
 }
 
-func (n *DocumentSet) AsBytesWithPrinter(printerFunc func(io.Writer) DocumentPrinter) ([]byte, error) {
+func (ds *DocumentSet) SetRawBytes(docSet *DocumentSet) {
+	// TODO: DocumentSet.originalBytes already contain this information
+	//       This sets RawData into the documents, but in the end it contains the full document set bits and not
+	//       a document specific raw data. Eventually it might be ok but we should consider this as it might be a problem
+	//       Some options that I see are:
+	//         rename RawData to originalBytesDocumentSet or something.
+	//         templating will split these originalBytes we have into per document string.
+	//       I think the second option would be the one I would like but I think it might be a little bit of a big change
+	//       because we will have to change the yaml parser to do this....
```

Comment:
🛑  This TODO needs to be resolved before this PR can be merged.

Reply:
Position now has the responsibility of connecting Nodes to their source. parse sets it up and the error messages can use it without needing to ask the Document for the content at that line.